dev/com.ibm.ws.cdi.ejb_fat/fat/src/com/ibm/ws/cdi/ejb/apps/newTimer/TestEjbTimerServlet.java

```diff
@@ -0,0 +1,48 @@
+ /*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.cdi.ejb.apps.newtimer;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+
+import javax.ejb.EJB;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import componenttest.app.FATServlet;
+import org.junit.Test;
+
+@WebServlet("/Timer")
+public class TestEjbTimerServlet extends FATServlet {
+    private static final long serialVersionUID = 1L;
+    @EJB
+    EjbSessionBeanLocal bean;
+
+    @Test
+    public void testEjbTimers() throws ServletException, IOException {
+        bean.initTimer();
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
```

Comment:
Can we avoid using `sleep`? It means we always have to wait the sleep time and there's a risk that something is slow and we don't wait long enough.

We could have a `CompletableFuture` stored in a static field which is completed by the timeout method. In this method we would wait on the `CompletableFuture` so that the rest of the test runs as soon as it completes.